lib/src/merge.rs

```diff
@@ -639,6 +641,20 @@ where
     }
 }
 
+impl<F, T, E> Merge<F>
+where
+    F: Future<Output = Result<T, E>>,
+{
+    /// Evaluates the futures in the input merge concurrently. Returns an error
+    /// if any of the futures returned an error.
+    pub async fn try_join_all(self) -> Result<Merge<T>, E> {
+        let values = try_join_all(self.values).await?;
+        Ok(Merge {
+            values: values.into(),
+        })
+    }
```

Comment:
nit: this means we have to create a temporary `Merge { values: Vec<F> }`. The cost wouldn't matter, but it seems wasteful. Maybe we can instead add `.try_map_async()`?

Reply:
Good idea. Done.